src/bin/uniq.rs

```diff
@@ -0,0 +1,171 @@
+#![deny(warnings)]
+
+extern crate coreutils;
+extern crate extra;
+
+use std::env;
+use std::fs::File;
+use std::process::exit;
+use std::io::{stdout, stderr, stdin, Error, Write, BufRead, BufReader};
+use coreutils::ArgParser;
+use extra::option::OptionalExt;
+
+const MAN_PAGE: &'static str = r#"
+NAME
+    uniq - report or omit repeated lines
+
+SYNOPSIS
+    uniq [ -h | --help | -d | -u | -i ] [FILE]...
+
+DESCRIPTION
+    Filter adjacent matching lines of FILE(s) to standard output.
+
+    With no FILE(s), read standard input.
+
+OPTIONS
+    -h
+    --help
+        display this help and exit
+    -c
+    --count
+        precede each output line with a count of the number of times the line occurred in the input
+    -i
+    --ignore-case
+        compare lines case-insensitively
+    -d
+    --repeated-lines
+        only print duplicate lines, one for each group
+    -u
+    --unique-lines
+        only print unique lines
+"#; /* @MANEND */
+
+fn lines_from_stdin() -> Result<Vec<String>, Error> {
+    let stdin = stdin();
+    let mut lines = Vec::new();
+
+    let f = BufReader::new(stdin.lock());
+    for line in f.lines() {
+        let line = match line {
+            Ok(l) => l,
+            Err(e) => return Err(e),
+        };
+        lines.push(line);
+    }
+    Ok(lines)
+}
+
+fn lines_from_files(paths: &Vec<&String>) -> Result<Vec<String>, Error> {
+    let mut lines = Vec::new();
+
+    for path in paths {
+        let f = try!(File::open(path));
+        let f = BufReader::new(f);
+        for line in f.lines() {
+            let line = match line {
+                Ok(l) => l,
+                Err(e) => return Err(e),
+            };
+            lines.push(line);
+        }
+    }
+    Ok(lines)
```

Comment:
Thanks, @mmstick. I learned a lot from your suggestions.

I wasn't able to implement this one, though. Apparently, the `?` doesn't work inside a closure and I had to use `unwrap()` instead, which seemed inappropriate.

Perhaps am I missing something?